refactor: centralize toast notifications into shared utility

antaripdebgupta opened the pull request:
### **Description**

Direct `toast.success / toast.error / toast.info / toast.warning` calls across components and composables have been replaced with shared helpers to reduce duplication and improve consistency.

No behavior or UI changes are intended.

**Out of scope:**

* Vuex-based toast dispatch (`SET_TOAST`)
* `GlobalToastHandler` and `GlobalErrorHandler`

---

### **Related Issue**

Fixes: #1078 

---

### **Context**

* Toast logic was duplicated across many files
* Inconsistent usage patterns (raw strings, i18n resolution, dynamic messages)
* Centralizing simplifies future changes and maintenance

---

### **Changes**

* Added shared toast utility
* Migrated direct toast calls to use the shared helpers
* Kept existing global toast handling intact

antaripdebgupta committed:
refactor: centralize toast notifications into shared utility

marcgc21 commented:
@antaripdebgupta It looks interesting to provide this layer. However right now we have conflicts. Can you solve them before we merge it?

antaripdebgupta commented:
@marcgc21 Sure, I’ll resolve the merge conflicts and update the PR.

antaripdebgupta committed:
Merge branch 'develop' into fix-issue-1078

antaripdebgupta committed:
resolve merge conflict in ProfileView.vue

marcgc21 commented:
@antaripdebgupta take a look on other places that still appears the vue-toastification. In theory if we are going to use this new util, the import it should be called once. 

`import { useToast } from 'vue-toastification';`

antaripdebgupta commented:
@marcgc21,
I’ll remove the remaining direct `useToast` imports and route everything through the shared toast util. Update coming soon.

antaripdebgupta committed:
refactor: update global handlers to use toast utility

antaripdebgupta commented:
@marcgc21,  I've addressed your feedback.

This completes the toast centralization refactor by ensuring all toast calls throughout the application flow through the shared utility, maintaining consistency and reducing duplication.

### Verification
`grep -RIn "useToast" src --include="*.js" --include="*.vue" --include="*.ts"`
```
src/main.js:7:import Toast, { useToast } from 'vue-toastification';
src/main.js:33:app.config.globalProperties.$toast = useToast();
src/shared/utils/toast.js:1:import { useToast } from 'vue-toastification'
src/shared/utils/toast.js:4:const toast = useToast()

```

The main.js imports are for plugin setup. All application code now uses the centralized utility.

marcgc21 commented:
Awesome, hope we don't break anything!
